feat: allow integrations to refresh content layer data (#11878)
withastro/astro: .changeset/curvy-walls-kneel.md

---
'astro': patch
---

> Adds support for allowing integrations to refresh the content layer
> 
> This adds a new object to the `astro:server:setup` hook that allows integrations to refresh the content layer. This can be used for example to register a webhook endpoint during dev, or to open a socket to a CMS to listen for changes.

Reviewer: ```suggestion
Adds a new object `refreshContent` to the `astro:server:setup` hook that allows integrations to refresh the content layer. This can be used, for example, to register a webhook endpoint during dev, or to open a socket to a CMS to listen for changes.
```
My preference is always to get the new thing named asap, so we could do something like this!

> The hook is passed a function called `refreshContent` that can be called to refresh the content layer. It can optionally be passed a `loaders` property, which is an array of loader names. If provided, only collections that use those loaders will be refreshed. If not provided, all loaders will be refreshed. A CMS integration could use this to only refresh its own collections.

Reviewer: ```suggestion
By default, `refreshContent` will refresh all collections. You can optionally pass a `loaders` property, which is an array of loader names. If provided, only collections that use those loaders will be refreshed. For example, A CMS integration could use this property to only refresh its own collections.
```

> It can also pass a `context` object, which will be passed to the loaders. This can be used to pass arbitrary data, such as the webhook body or an event from the websocket.

Reviewer: ```suggestion
You can also pass a `context` object to the loaders. This can be used to pass arbitrary data, such as the webhook body, or an event from the websocket.
```

```ts
 {
		name: 'my-integration',

> 		hooks: {

Reviewer 2: nit: I think we use spaces in markdown files
